diff --git a/php-src/ext/mbstring/libmbfl/filters/mbfilter_utf8_mobile.c b/php-src/ext/mbstring/libmbfl/filters/mbfilter_utf8_mobile.c
--- a/php-src/ext/mbstring/libmbfl/filters/mbfilter_utf8_mobile.c	2023-01-18 15:22:55
+++ b/php-src/ext/mbstring/libmbfl/filters/mbfilter_utf8_mobile.c	2023-01-18 15:22:57
@@ -39,8 +39,9 @@
 extern int mbfl_filt_ident_utf8(int c, mbfl_identify_filter *filter);
 extern int mbfl_filt_conv_utf8_wchar_flush(mbfl_convert_filter *filter);
+extern int mbfl_filt_put_invalid_char(int c, mbfl_convert_filter *filter);

 extern const unsigned char mblen_table_utf8[];

 static const char *mbfl_encoding_utf8_docomo_aliases[] = {"UTF-8-DOCOMO", "UTF8-DOCOMO", NULL};
 static const char *mbfl_encoding_utf8_kddi_a_aliases[] = {NULL};
 static const char *mbfl_encoding_utf8_kddi_b_aliases[] = {"UTF-8-Mobile#KDDI", "UTF-8-KDDI", "UTF8-KDDI", NULL};
diff --git a/php-src/ext/pcre/pcrelib/config.h b/php-src/ext/pcre/pcrelib/config.h
index 0f7a9f73ff..b67dcfa94d 100644
--- a/php-src/ext/pcre/pcrelib/config.h
+++ b/php-src/ext/pcre/pcrelib/config.h
@@ -397,7 +397,7 @@ them both to 0; an emulation function will be used. */
 #undef SUPPORT_GCOV
 
 /* Define to any value to enable support for Just-In-Time compiling. */
-#define SUPPORT_JIT
+#undef SUPPORT_JIT
 
 /* Define to any value to allow pcregrep to be linked with libbz2, so that it
    is able to handle .bz2 files. */
diff --git a/php-src/ext/pcre/pcrelib/sljit/sljitLir.c b/php-src/ext/pcre/pcrelib/sljit/sljitLir.c
index 0f1b1c9cce..2263f97574 100644
--- a/php-src/ext/pcre/pcrelib/sljit/sljitLir.c
+++ b/php-src/ext/pcre/pcrelib/sljit/sljitLir.c
@@ -1715,12 +1715,6 @@ SLJIT_API_FUNC_ATTRIBUTE SLJIT_CONST char* sljit_get_platform_name(void)
 	return "unsupported";
 }
 
-SLJIT_API_FUNC_ATTRIBUTE struct sljit_compiler* sljit_create_compiler(void)
-{
-	SLJIT_ASSERT_STOP();
-	return NULL;
-}
-
 SLJIT_API_FUNC_ATTRIBUTE void sljit_free_compiler(struct sljit_compiler *compiler)
 {
 	SLJIT_UNUSED_ARG(compiler);
diff --git a/php-src/ext/pcre/pcrelib/sljit/sljitConfigInternal.h b/php-src/ext/pcre/pcrelib/sljit/sljitConfigInternal.h
index 16e3547c93..1b7934f095 100644
--- a/php-src/ext/pcre/pcrelib/sljit/sljitConfigInternal.h
+++ b/php-src/ext/pcre/pcrelib/sljit/sljitConfigInternal.h
@@ -126,7 +126,7 @@
 #else
 #define SLJIT_CONFIG_ARM_V5 1
 #endif
-#elif defined (__aarch64__)
+#elif defined(__aarch64__) || defined(__arm64__) || defined(__arm64) || defined(__aarch32__)
 #define SLJIT_CONFIG_ARM_64 1
 #elif defined(__ppc64__) || defined(__powerpc64__) || defined(_ARCH_PPC64) || (defined(_POWER) && defined(__64BIT__))
 #define SLJIT_CONFIG_PPC_64 1

diff --git a/php-src/main/streams/cast.c b/php-src/main/streams/cast.c
index 0603a7212b..2c191bc8be 100644
--- a/php-src/main/streams/cast.c
+++ b/php-src/main/streams/cast.c
@@ -122,11 +122,11 @@ static int stream_cookie_seeker(void *cookie, __off64_t *position, int whence)
 	return 0;
 }
 # else
-static int stream_cookie_seeker(void *cookie, off_t position, int whence)
+static int stream_cookie_seeker(void *cookie, off_t *position, int whence)
 {
 	TSRMLS_FETCH();
 
-	return php_stream_seek((php_stream *)cookie, position, whence);
+	return php_stream_seek((php_stream *)cookie, *position, whence);
 }
 # endif
 
