diff --git a/php-src/ext/standard/file.c b/php-src/ext/standard/file.c
--- a/php-src/ext/standard/file.c
+++ b/php-src/ext/standard/file.c
@@ -1691,18 +1691,32 @@ PHPAPI int php_copy_file_ctx(const char *src, const char *dest, int src_flg, php
 	php_stream *srcstream = NULL, *deststream = NULL;
 	int ret = FAILURE;
 	php_stream_statbuf src_s, dest_s;
 
 	switch (php_stream_stat_path_ex(src, 0, &src_s, ctx)) {
 		case -1:
 			/* non-statable stream */
 			goto safe_to_copy;
 			break;
 		case 0:
+			// Fix for https://github.com/WordPress/wordpress-playground/issues/54:
+			// Problem: Calling copy() on an empty source file crashes the JavaScript
+			//          runtime.
+			// Solution: Avoid copying empty files. Just create create an empty
+			//           destination file and return.
+			if (src_s.sb.st_size == 0) {
+				zend_string *opened_path = zend_string_init("", strlen(""), 0);
+				php_stream *stream = php_stream_open_wrapper(dest, "w", REPORT_ERRORS, &opened_path);
+				if (stream) {
+					php_stream_close(stream);
+					return SUCCESS;
+				}
+				return FAILURE;
+			}
 			break;
 		default: /* failed to stat file, does not exist? */
 			return ret;
 	}
 	if (S_ISDIR(src_s.sb.st_mode)) {
 		php_error_docref(NULL, E_WARNING, "The first argument to copy() function cannot be a directory");
 		return FAILURE;
 	}
diff --git a/php-src/ext/date/lib/timelib.h b/php-src/ext/date/lib/timelib.h
index 54b9287b5d..2a4bf7b0cc 100644
--- a/php-src/ext/date/lib/timelib.h
+++ b/php-src/ext/date/lib/timelib.h
@@ -105,7 +105,7 @@ typedef unsigned __int64  uint64_t;
 # endif
 #endif
 
-#if (defined(__x86_64__) || defined(__LP64__) || defined(_LP64) || defined(_WIN64)) && !defined(TIMELIB_FORCE_LONG32)
+#if true
 typedef int64_t timelib_long;
 typedef uint64_t timelib_ulong;
 # define TIMELIB_LONG_MAX INT64_MAX
@@ -123,10 +123,10 @@ typedef uint32_t timelib_ulong;
 # define TIMELIB_ULONG_FMT "%" PRIu32
 #endif
 
-#if defined(_MSC_VER)
+#if true
 typedef uint64_t timelib_ull;
 typedef int64_t timelib_sll;
-# define TIMELIB_LL_CONST(n) n ## i64
+# define TIMELIB_LL_CONST(n) n ## ll
 #else
 typedef unsigned long long timelib_ull;
 typedef signed long long timelib_sll;
diff --git a/php-src/ext/date/php_date.c b/php-src/ext/date/php_date.c
index 082b475e43..d317f82388 100644
--- a/php-src/ext/date/php_date.c
+++ b/php-src/ext/date/php_date.c
@@ -1018,7 +1018,8 @@ PHP_FUNCTION(strtotime)
 	zend_string *times;
 	int parse_error, epoch_does_not_fit_in_zend_long;
 	timelib_error_container *error;
-	zend_long preset_ts, ts;
+	zend_long preset_ts;
+	double ts;
 	zend_bool preset_ts_is_null = 1;
 	timelib_time *t, *now;
 	timelib_tzinfo *tzi;
@@ -1057,7 +1058,7 @@ PHP_FUNCTION(strtotime)
 
 	timelib_fill_holes(t, now, TIMELIB_NO_CLONE);
 	timelib_update_ts(t, tzi);
-	ts = timelib_date_to_int(t, &epoch_does_not_fit_in_zend_long);
+	ts = (double)timelib_date_to_int(t, &epoch_does_not_fit_in_zend_long);
 
 	timelib_time_dtor(now);
 	timelib_time_dtor(t);
@@ -1067,7 +1068,7 @@ PHP_FUNCTION(strtotime)
 		RETURN_FALSE;
 	}
 
-	RETURN_LONG(ts);
+	RETURN_DOUBLE(ts);
 }
 /* }}} */
 
diff --git a/php-src/ext/date/php_date_arginfo.h b/php-src/ext/date/php_date_arginfo.h
index cf50abb43f..284d1080b2 100644
--- a/php-src/ext/date/php_date_arginfo.h
+++ b/php-src/ext/date/php_date_arginfo.h
@@ -1,9 +1,9 @@
 /* This is a generated file, edit the .stub.php file instead.
  * Stub hash: 880a93d8a3461635447318317869fa7d3d7762dd */
 
-ZEND_BEGIN_ARG_WITH_RETURN_TYPE_MASK_EX(arginfo_strtotime, 0, 1, MAY_BE_LONG|MAY_BE_FALSE)
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_MASK_EX(arginfo_strtotime, 0, 1, MAY_BE_DOUBLE|MAY_BE_FALSE)
 	ZEND_ARG_TYPE_INFO(0, datetime, IS_STRING, 0)
-	ZEND_ARG_TYPE_INFO_WITH_DEFAULT_VALUE(0, baseTimestamp, IS_LONG, 1, "null")
+	ZEND_ARG_TYPE_INFO_WITH_DEFAULT_VALUE(0, baseTimestamp, IS_DOUBLE, 1, "null")
 ZEND_END_ARG_INFO()
 
 ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(arginfo_date, 0, 1, IS_STRING, 0)
