From a7cc5e7c725651dbf39fd8f90c8374cbcc1a5bec Mon Sep 17 00:00:00 2001
From: port <port@example.invalid>
Date: Sat, 18 Jul 2026 04:28:21 +0200
Subject: [PATCH 03/11] Use WASI randomness for PHP CSPRNG

---
 ext/standard/random.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/standard/random.c b/ext/standard/random.c
index a023a0f9..cc62618f 100644
--- a/ext/standard/random.c
+++ b/ext/standard/random.c
@@ -90,7 +90,9 @@ PHP_MSHUTDOWN_FUNCTION(random)
 /* {{{ php_random_bytes */
 PHPAPI int php_random_bytes(void *bytes, size_t size, bool should_throw)
 {
-#ifdef PHP_WIN32
+#ifdef PHP_WASI
+	arc4random_buf(bytes, size);
+#elif defined(PHP_WIN32)
 	/* Defer to CryptGenRandom on Windows */
 	if (php_win32_get_random_bytes(bytes, size) == FAILURE) {
 		if (should_throw) {
-- 
2.50.1

