From 9a6168a71a3c88f92531a511bafef882b6b7c20e Mon Sep 17 00:00:00 2001
From: port <port@example.invalid>
Date: Sat, 18 Jul 2026 04:29:33 +0200
Subject: [PATCH 09/11] Port WASI compatibility patch 0010

---
 main/streams/xp_socket.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/main/streams/xp_socket.c b/main/streams/xp_socket.c
index 484c9d3f..525caa4d 100644
--- a/main/streams/xp_socket.c
+++ b/main/streams/xp_socket.c
@@ -401,7 +401,12 @@ static int php_sockop_set_option(php_stream *stream, int option, int value, void
 				case STREAM_XPORT_OP_SEND:
 					flags = 0;
 					if ((xparam->inputs.flags & STREAM_OOB) == STREAM_OOB) {
+#ifdef MSG_OOB
 						flags |= MSG_OOB;
+#else
+						xparam->outputs.returncode = -1;
+						return PHP_STREAM_OPTION_RETURN_OK;
+#endif
 					}
 					xparam->outputs.returncode = sock_sendto(sock,
 							xparam->inputs.buf, xparam->inputs.buflen,
@@ -419,7 +424,12 @@ static int php_sockop_set_option(php_stream *stream, int option, int value, void
 				case STREAM_XPORT_OP_RECV:
 					flags = 0;
 					if ((xparam->inputs.flags & STREAM_OOB) == STREAM_OOB) {
+#ifdef MSG_OOB
 						flags |= MSG_OOB;
+#else
+						xparam->outputs.returncode = -1;
+						return PHP_STREAM_OPTION_RETURN_OK;
+#endif
 					}
 					if ((xparam->inputs.flags & STREAM_PEEK) == STREAM_PEEK) {
 						flags |= MSG_PEEK;
-- 
2.50.1

