From 91f7a4a3e3180bfd58622fece82fa5df14ceef1a Mon Sep 17 00:00:00 2001
From: WordPress Playground contributors <playground@wordpress.org>
Date: Sat, 18 Jul 2026 04:40:31 +0200
Subject: [PATCH 11/12] Avoid legacy hostname lookups on WASI

---
 main/network.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/main/network.c b/main/network.c
index b54f2c1d..fe6b6112 100644
--- a/main/network.c
+++ b/main/network.c
@@ -1336,7 +1336,7 @@ PHPAPI int php_poll2(php_pollfd *ufds, unsigned int nfds, int timeout)
 }
 #endif
 
-#if defined(HAVE_GETHOSTBYNAME_R)
+#if defined(HAVE_GETHOSTBYNAME_R) && !defined(PHP_WASI)
 #ifdef HAVE_FUNC_GETHOSTBYNAME_R_6
 static struct hostent * gethostname_re (const char *host,struct hostent *hostbuf,char **tmphstbuf,size_t *hstbuflen)
 {
@@ -1408,7 +1408,9 @@ static struct hostent * gethostname_re (const char *host,struct hostent *hostbuf
 #endif
 
 PHPAPI struct hostent*	php_network_gethostbyname(const char *name) {
-#if !defined(HAVE_GETHOSTBYNAME_R)
+#if defined(PHP_WASI)
+	return NULL;
+#elif !defined(HAVE_GETHOSTBYNAME_R)
 	return gethostbyname(name);
 #else
 	if (FG(tmp_host_buf)) {
-- 
2.50.1

