From 156d8cc9f369b4db90963daf88dbdc9e6b3492d4 Mon Sep 17 00:00:00 2001
From: port <port@example.invalid>
Date: Sat, 18 Jul 2026 04:43:50 +0200
Subject: [PATCH 10/11] Avoid legacy hostname lookups on WASI

---
 main/network.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/main/network.c b/main/network.c
index f6b9a2d1..da51f2db 100644
--- a/main/network.c
+++ b/main/network.c
@@ -1248,7 +1248,7 @@ PHPAPI int php_poll2(php_pollfd *ufds, unsigned int nfds, int timeout)
 }
 #endif
 
-#if defined(HAVE_GETHOSTBYNAME_R)
+#if defined(HAVE_GETHOSTBYNAME_R) && !defined(PHP_WASI)
 #ifdef HAVE_FUNC_GETHOSTBYNAME_R_6
 struct hostent * gethostname_re (const char *host,struct hostent *hostbuf,char **tmphstbuf,size_t *hstbuflen)
 {
@@ -1320,7 +1320,9 @@ struct hostent * gethostname_re (const char *host,struct hostent *hostbuf,char *
 #endif
 
 PHPAPI struct hostent*	php_network_gethostbyname(char *name) {
-#if !defined(HAVE_GETHOSTBYNAME_R)
+#if defined(PHP_WASI)
+	return NULL;
+#elif !defined(HAVE_GETHOSTBYNAME_R)
 	return gethostbyname(name);
 #else
 	if (FG(tmp_host_buf)) {
-- 
2.50.1

