diff --git a/main/network.c b/main/network.c
index 62bcf5db41..78e2371f83 100644
--- a/main/network.c
+++ b/main/network.c
@@ -66,7 +66,7 @@ int inet_aton(const char *, struct in_addr *);
 
 #include "php_network.h"
 
-#if defined(PHP_WIN32) || defined(__riscos__)
+#if defined(PHP_WIN32) || defined(__riscos__) || defined(PHP_WASI)
 #undef AF_UNIX
 #endif
 
diff --git a/main/streams/streams.c b/main/streams/streams.c
index 6388ac7a9c..07c0065821 100644
--- a/main/streams/streams.c
+++ b/main/streams/streams.c
@@ -1832,6 +1832,6 @@ int php_init_stream_wrappers(int module_number)
 			&&
 			php_stream_xport_register("udp", php_stream_generic_socket_factory) == SUCCESS
-#if defined(AF_UNIX) && !(defined(PHP_WIN32) || defined(__riscos__))
+#if defined(AF_UNIX) && !(defined(PHP_WIN32) || defined(__riscos__) || defined(PHP_WASI))
 			&&
 			php_stream_xport_register("unix", php_stream_generic_socket_factory) == SUCCESS
 			&&
diff --git a/main/streams/xp_socket.c b/main/streams/xp_socket.c
index a8c2dfc80e..eb0dd25bf6 100644
--- a/main/streams/xp_socket.c
+++ b/main/streams/xp_socket.c
@@ -18,7 +18,7 @@
 #include "streams/php_streams_int.h"
 #include "php_network.h"
 
-#if defined(PHP_WIN32) || defined(__riscos__)
+#if defined(PHP_WIN32) || defined(__riscos__) || defined(PHP_WASI)
 # undef AF_UNIX
 #endif
 
