diff --git a/php-src/Zend/zend_alloc.c b/php-src/Zend/zend_alloc.c
index bf2116fc91..bec65453d8 100644
--- a/php-src/Zend/zend_alloc.c
+++ b/php-src/Zend/zend_alloc.c
@@ -773,7 +773,7 @@ static void *zend_mm_chunk_alloc(zend_mm_heap *heap, size_t size, size_t alignme
 #if ZEND_MM_STORAGE
 	if (UNEXPECTED(heap->storage)) {
 		void *ptr = heap->storage->handlers.chunk_alloc(heap->storage, size, alignment);
-		ZEND_ASSERT(((zend_uintptr_t)((char*)ptr + (alignment-1)) & (alignment-1)) == (zend_uintptr_t)ptr);
+		ZEND_ASSERT(((zend_uintptr_t)((char*)ptr + (alignment-1)) & ~(alignment-1)) == (zend_uintptr_t)ptr);
 		return ptr;
 	}
 #endif
