diff --git a/php-src/ext/opcache/zend_accelerator_debug.c b/php-src/ext/opcache/zend_accelerator_debug.c
index 4932999b624..c3f034855b5 100644
--- a/php-src/ext/opcache/zend_accelerator_debug.c
+++ b/php-src/ext/opcache/zend_accelerator_debug.c
@@ -25,6 +25,9 @@
 #include <time.h>
 #ifdef ZEND_WIN32
 # include <process.h>
+#else
+/* Error: call to undeclared function 'getpid' */
+# include <unistd.h>
 #endif
 #include "ZendAccelerator.h"

diff --git a/php-src/ext/pdo_sqlite/sqlite_statement.c b/php-src/ext/pdo_sqlite/sqlite_statement.c
index 817e6294db1..2a30874f1c5 100644
--- a/php-src/ext/pdo_sqlite/sqlite_statement.c
+++ b/php-src/ext/pdo_sqlite/sqlite_statement.c
@@ -261,7 +261,7 @@ static int pdo_sqlite_stmt_describe(pdo_stmt_t *stmt, int colno)
 	return 1;
 }

-static int pdo_sqlite_stmt_get_col(pdo_stmt_t *stmt, int colno, char **ptr, zend_ulong *len, int *caller_frees)
+static int pdo_sqlite_stmt_get_col(pdo_stmt_t *stmt, int colno, char **ptr, size_t *len, int *caller_frees)
 {
 	pdo_sqlite_stmt *S = (pdo_sqlite_stmt*)stmt->driver_data;
 	if (!S->stmt) {
diff --git a/php-src/ext/standard/file.c b/php-src/ext/standard/file.c
index 0bb81cae8ba..f5ffbe48b65 100644
--- a/php-src/ext/standard/file.c
+++ b/php-src/ext/standard/file.c
@@ -1666,6 +1666,20 @@ PHPAPI int php_copy_file_ctx(const char *src, const char *dest, int src_flg, php
 			goto safe_to_copy;
 			break;
 		case 0:
+			// Fix for https://github.com/WordPress/wordpress-playground/issues/54:
+			// Problem: Calling copy() on an empty source file crashes the JavaScript
+			//          runtime.
+			// Solution: Avoid copying empty files. Just create create an empty
+			//           destination file and return.
+			if (src_s.sb.st_size == 0) {
+				zend_string *opened_path = zend_string_init("", strlen(""), 0);
+				php_stream *stream = php_stream_open_wrapper(dest, "w", REPORT_ERRORS, &opened_path);
+				if (stream) {
+					php_stream_close(stream);
+					return SUCCESS;
+				}
+				return FAILURE;
+			}
 			break;
 		default: /* failed to stat file, does not exist? */
 			return ret;
diff --git a/php-src/main/streams/cast.c b/php-src/main/streams/cast.c
index fe56c9b0b5d..ea3fd445911 100644
--- a/php-src/main/streams/cast.c
+++ b/php-src/main/streams/cast.c
@@ -115,10 +115,10 @@ static int stream_cookie_seeker(void *cookie, __off64_t *position, int whence)
 	return 0;
 }
 # else
-static int stream_cookie_seeker(void *cookie, zend_off_t position, int whence)
+static int stream_cookie_seeker(void *cookie, long long *position, int whence)
 {

-	return php_stream_seek((php_stream *)cookie, position, whence);
+	return php_stream_seek((php_stream *)cookie, *position, whence);
 }
 # endif

