diff --git a/php-src/ext/standard/file.c b/php-src/ext/standard/file.c
--- a/php-src/ext/standard/file.c
+++ b/php-src/ext/standard/file.c
@@ -1691,20 +1691,34 @@ PHPAPI int php_copy_file_ctx(const char *src, const char *dest, int src_flg, php
 	php_stream *srcstream = NULL, *deststream = NULL;
 	int ret = FAILURE;
 	php_stream_statbuf src_s, dest_s;

 	switch (php_stream_stat_path_ex(src, 0, &src_s, ctx)) {
 		case -1:
 			/* non-statable stream */
 			goto safe_to_copy;
 			break;
 		case 0:
+			// Fix for https://github.com/WordPress/wordpress-playground/issues/54:
+			// Problem: Calling copy() on an empty source file crashes the JavaScript
+			//          runtime.
+			// Solution: Avoid copying empty files. Just create create an empty
+			//           destination file and return.
+			if (src_s.sb.st_size == 0) {
+				zend_string *opened_path = zend_string_init("", strlen(""), 0);
+				php_stream *stream = php_stream_open_wrapper(dest, "w", REPORT_ERRORS, &opened_path);
+				if (stream) {
+					php_stream_close(stream);
+					return SUCCESS;
+				}
+				return FAILURE;
+			}
 			break;
 		default: /* failed to stat file, does not exist? */
 			return ret;
 	}
 	if (S_ISDIR(src_s.sb.st_mode)) {
 		php_error_docref(NULL, E_WARNING, "The first argument to copy() function cannot be a directory");
 		return FAILURE;
 	}

 	switch (php_stream_stat_path_ex(dest, PHP_STREAM_URL_STAT_QUIET | PHP_STREAM_URL_STAT_NOCACHE, &dest_s, ctx)) {


diff --git a/php-src/Zend/zend_variables.c b/php-src/Zend/zend_variables.c
index 810866a1be2..0cc2a95dcf2 100644
--- a/php-src/Zend/zend_variables.c
+++ b/php-src/Zend/zend_variables.c
@@ -36,6 +36,16 @@ static void ZEND_FASTCALL zend_empty_destroy(zend_reference *ref);

 typedef void (ZEND_FASTCALL *zend_rc_dtor_func_t)(zend_refcounted *p);

+// Note: zend_list_free_wrapper function is required because active streams
+// would otherwise cause a function signature mismatch. zend_list_free
+// returns an int, while zend_rc_dtor_func_t expects a void return type.
+static void zend_list_free_wrapper(zend_refcounted *p) {
+	zend_resource *res = (zend_resource*)p;
+	zval zv;
+	ZVAL_RES(&zv, res);
+	zend_list_free(&zv);
+}
+
 static const zend_rc_dtor_func_t zend_rc_dtor_func[] = {
 	/* IS_UNDEF        */ (zend_rc_dtor_func_t)zend_empty_destroy,
 	/* IS_NULL         */ (zend_rc_dtor_func_t)zend_empty_destroy,
@@ -46,7 +56,7 @@ static const zend_rc_dtor_func_t zend_rc_dtor_func[] = {
 	/* IS_STRING       */ (zend_rc_dtor_func_t)zend_string_destroy,
 	/* IS_ARRAY        */ (zend_rc_dtor_func_t)zend_array_destroy,
 	/* IS_OBJECT       */ (zend_rc_dtor_func_t)zend_objects_store_del,
-	/* IS_RESOURCE     */ (zend_rc_dtor_func_t)zend_list_free,
+	/* IS_RESOURCE     */ zend_list_free_wrapper,
 	/* IS_REFERENCE    */ (zend_rc_dtor_func_t)zend_reference_destroy,
 	/* IS_CONSTANT_AST */ (zend_rc_dtor_func_t)zend_ast_ref_destroy
 };
