diff --git a/php-src/Zend/zend_variables.c b/php-src/Zend/zend_variables.c
index 810866a1be2..279c49e04fc 100644
--- a/php-src/Zend/zend_variables.c
+++ b/php-src/Zend/zend_variables.c
@@ -36,6 +36,14 @@ static void ZEND_FASTCALL zend_empty_destroy(zend_reference *ref);

 typedef void (ZEND_FASTCALL *zend_rc_dtor_func_t)(zend_refcounted *p);

+// Note: zend_list_free_wrapper function is required because active streams
+// would otherwise cause a function signature mismatch. zend_list_free
+// returns an int, while zend_rc_dtor_func_t expects a void return type.
+static void zend_list_free_wrapper(zend_refcounted *p) {
+	zend_resource *res = (zend_resource*)p;
+	zend_list_free(res);
+}
+
 static const zend_rc_dtor_func_t zend_rc_dtor_func[] = {
 	/* IS_UNDEF        */ (zend_rc_dtor_func_t)zend_empty_destroy,
 	/* IS_NULL         */ (zend_rc_dtor_func_t)zend_empty_destroy,
@@ -46,7 +54,7 @@ static const zend_rc_dtor_func_t zend_rc_dtor_func[] = {
 	/* IS_STRING       */ (zend_rc_dtor_func_t)zend_string_destroy,
 	/* IS_ARRAY        */ (zend_rc_dtor_func_t)zend_array_destroy,
 	/* IS_OBJECT       */ (zend_rc_dtor_func_t)zend_objects_store_del,
-	/* IS_RESOURCE     */ (zend_rc_dtor_func_t)zend_list_free,
+	/* IS_RESOURCE     */ zend_list_free_wrapper,
 	/* IS_REFERENCE    */ (zend_rc_dtor_func_t)zend_reference_destroy,
 	/* IS_CONSTANT_AST */ (zend_rc_dtor_func_t)zend_ast_ref_destroy
 };

diff --git a/php-src/ext/opcache/zend_accelerator_debug.c b/php-src/ext/opcache/zend_accelerator_debug.c
index 991f2329da9..b19a12481c2 100644
--- a/php-src/ext/opcache/zend_accelerator_debug.c
+++ b/php-src/ext/opcache/zend_accelerator_debug.c
@@ -25,6 +25,9 @@
 #include <time.h>
 #ifdef ZEND_WIN32
 # include <process.h>
+#else
+/* Error: call to undeclared function 'getpid' */
+# include <unistd.h>
 #endif
 #include "ZendAccelerator.h"

diff --git a/php-src/ext/standard/file.c b/php-src/ext/standard/file.c
index 3bd3421603f..6ef5cae9635 100644
--- a/php-src/ext/standard/file.c
+++ b/php-src/ext/standard/file.c
@@ -1698,6 +1698,20 @@ PHPAPI int php_copy_file_ctx(const char *src, const char *dest, int src_flg, php
 			goto safe_to_copy;
 			break;
 		case 0:
+			// Fix for https://github.com/WordPress/wordpress-playground/issues/54:
+			// Problem: Calling copy() on an empty source file crashes the JavaScript
+			//          runtime.
+			// Solution: Avoid copying empty files. Just create create an empty
+			//           destination file and return.
+			if (src_s.sb.st_size == 0) {
+				zend_string *opened_path = zend_string_init("", strlen(""), 0);
+				php_stream *stream = php_stream_open_wrapper(dest, "w", REPORT_ERRORS, &opened_path);
+				if (stream) {
+					php_stream_close(stream);
+					return SUCCESS;
+				}
+				return FAILURE;
+			}
 			break;
 		default: /* failed to stat file, does not exist? */
 			return ret;
