diff --git a/php-src/Zend/zend_variables.c b/php-src/Zend/zend_variables.c
index c2594274dd4..2a3c2a7a120 100644
--- a/php-src/Zend/zend_variables.c
+++ b/php-src/Zend/zend_variables.c
@@ -44,6 +44,14 @@ static void ZEND_FASTCALL zend_ast_ref_destroy_wrapper(zend_ast_ref *ast);

 typedef void (ZEND_FASTCALL *zend_rc_dtor_func_t)(zend_refcounted *p);

+// Note: zend_list_free_wrapper function is required because active streams
+// would otherwise cause a function signature mismatch. zend_list_free
+// returns an int, while zend_rc_dtor_func_t expects a void return type.
+static void zend_list_free_wrapper(zend_refcounted *p) {
+	zend_resource *res = (zend_resource*)p;
+	zend_list_free(res);
+}
+
 static const zend_rc_dtor_func_t zend_rc_dtor_func[] = {
 	/* IS_UNDEF        */ (zend_rc_dtor_func_t)zend_empty_destroy,
 	/* IS_NULL         */ (zend_rc_dtor_func_t)zend_empty_destroy,
@@ -54,7 +62,9 @@ static const zend_rc_dtor_func_t zend_rc_dtor_func[] = {
 	/* IS_STRING       */ (zend_rc_dtor_func_t)zend_string_destroy,
 	/* IS_ARRAY        */ (zend_rc_dtor_func_t)zend_array_destroy_wrapper,
 	/* IS_OBJECT       */ (zend_rc_dtor_func_t)zend_object_destroy_wrapper,
-	/* IS_RESOURCE     */ (zend_rc_dtor_func_t)zend_resource_destroy_wrapper,
+	// Note: zend_resource_destroy_wrapper is an alias for zend_list_free
+	// https://github.com/php/php-src/blob/PHP-7.3.33/Zend/zend_variables.c#L41
+	/* IS_RESOURCE     */ zend_list_free_wrapper,
 	/* IS_REFERENCE    */ (zend_rc_dtor_func_t)zend_reference_destroy,
 	/* IS_CONSTANT_AST */ (zend_rc_dtor_func_t)zend_ast_ref_destroy_wrapper
 };

diff --git a/php-src/main/streams/cast.c b/php-src/main/streams/cast.c
index 6cfa6519779..6a630db8511 100644
--- a/php-src/main/streams/cast.c
+++ b/php-src/main/streams/cast.c
@@ -113,10 +113,10 @@ static int stream_cookie_seeker(void *cookie, __off64_t *position, int whence)
 	return 0;
 }
 # else
-static int stream_cookie_seeker(void *cookie, zend_off_t position, int whence)
+static int stream_cookie_seeker(void *cookie, long long *position, int whence)
 {

-	return php_stream_seek((php_stream *)cookie, position, whence);
+	return php_stream_seek((php_stream *)cookie, *position, whence);
 }
 # endif

diff --git a/php-src/ext/opcache/zend_accelerator_debug.c b/php-src/ext/opcache/zend_accelerator_debug.c
index bd0a5bdb0a4..47f723d34dc 100644
--- a/php-src/ext/opcache/zend_accelerator_debug.c
+++ b/php-src/ext/opcache/zend_accelerator_debug.c
@@ -25,6 +25,9 @@
 #include <time.h>
 #ifdef ZEND_WIN32
 # include <process.h>
+#else
+/* Error: call to undeclared function 'getpid' */
+# include <unistd.h>
 #endif
 #include "ZendAccelerator.h"

diff --git a/php-src/ext/pdo_sqlite/sqlite_statement.c b/php-src/ext/pdo_sqlite/sqlite_statement.c
index 2ef6c2ab836..9c3b4f7b2ce 100644
--- a/php-src/ext/pdo_sqlite/sqlite_statement.c
+++ b/php-src/ext/pdo_sqlite/sqlite_statement.c
@@ -299,7 +299,7 @@ static int pdo_sqlite_stmt_describe(pdo_stmt_t *stmt, int colno)
 	return 1;
 }

-static int pdo_sqlite_stmt_get_col(pdo_stmt_t *stmt, int colno, char **ptr, zend_ulong *len, int *caller_frees)
+static int pdo_sqlite_stmt_get_col(pdo_stmt_t *stmt, int colno, char **ptr, size_t *len, int *caller_frees)
 {
 	pdo_sqlite_stmt *S = (pdo_sqlite_stmt*)stmt->driver_data;
 	if (!S->stmt) {
diff --git a/php-src/ext/standard/file.c b/php-src/ext/standard/file.c
index dca9cf4fe33..5ebc935b496 100644
--- a/php-src/ext/standard/file.c
+++ b/php-src/ext/standard/file.c
@@ -1713,6 +1713,20 @@ PHPAPI int php_copy_file_ctx(const char *src, const char *dest, int src_flg, php
 			goto safe_to_copy;
 			break;
 		case 0:
+			// Fix for https://github.com/WordPress/wordpress-playground/issues/54:
+			// Problem: Calling copy() on an empty source file crashes the JavaScript
+			//          runtime.
+			// Solution: Avoid copying empty files. Just create create an empty
+			//           destination file and return.
+			if (src_s.sb.st_size == 0) {
+				zend_string *opened_path = zend_string_init("", strlen(""), 0);
+				php_stream *stream = php_stream_open_wrapper(dest, "w", REPORT_ERRORS, &opened_path);
+				if (stream) {
+					php_stream_close(stream);
+					return SUCCESS;
+				}
+				return FAILURE;
+			}
 			break;
 		default: /* failed to stat file, does not exist? */
 			return ret;
