From 99720344734cb0684904f497c61c22b81eb1a6d0 Mon Sep 17 00:00:00 2001
From: WordPress Playground contributors <playground@wordpress.org>
Date: Sat, 18 Jul 2026 03:57:27 +0200
Subject: [PATCH 05/10] Expose the real CLI SAPI to the component

---
 sapi/cli/php_cli.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/sapi/cli/php_cli.c b/sapi/cli/php_cli.c
index 3f027736..8934d4b5 100644
--- a/sapi/cli/php_cli.c
+++ b/sapi/cli/php_cli.c
@@ -1161,11 +1161,15 @@ static int do_cli(int argc, char **argv) /* {{{ */
 /* }}} */
 
 /* {{{ main */
+#ifdef PHP_WASI_COMPONENT_CLI
+int wordpress_php_wasi_cli_main(int argc, char *argv[])
+#else
 #ifdef PHP_CLI_WIN32_NO_CONSOLE
 int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nShowCmd)
 #else
 int main(int argc, char *argv[])
 #endif
+#endif
 {
 #if defined(PHP_WIN32)
 # ifdef PHP_CLI_WIN32_NO_CONSOLE
@@ -1374,6 +1378,13 @@ int main(int argc, char *argv[])
 	 * exiting.
 	 */
 	cleanup_ps_args(argv);
+#ifdef PHP_WASI_COMPONENT_CLI
+	if (fflush(NULL) == EOF && exit_status == SUCCESS) {
+		exit_status = 1;
+	}
+	return exit_status;
+#else
 	exit(exit_status);
+#endif
 }
 /* }}} */
-- 
2.50.1

